clients/src/test/java/org/apache/kafka/common/requests/OffsetCommitResponseTest.java

```diff
@@ -85,19 +96,186 @@ public void testParse() {
+        NameAndId topic5 = new NameAndId("topic5");
+        NameAndId topic6 = new NameAndId("topic6");
+
+        Map<String, Uuid> topicIds = new HashMap<>();
+        topicIds.put(topicOne, topic1Id);
+        asList(topic3, topic4, topic5, topic6).forEach(nai -> topicIds.put(nai.name, nai.id));
+
+        OffsetCommitResponse.Builder builder = new OffsetCommitResponse.Builder()
+            .addPartition(topicOne, topic1Id, partitionOne, Errors.NONE)
+            .addPartition(topicOne, topic1Id, partitionTwo, Errors.NONE)
+            .addPartitions(topic6.name, topic6.id, asList(11, 12), identity(), Errors.NONE);
+
+        OffsetCommitResponseData coordinatorResults = new OffsetCommitResponseData()
+            .setTopics(Arrays.asList(
+                createResponseTopic(topicTwo, Uuid.ZERO_UUID, 3, 4, Errors.NONE),
+                createResponseTopic(topic3.name, topic3.id, 5, 6, Errors.NONE),
+                createResponseTopic(topic4.name, topic4.id, 7, 8, Errors.NONE),
+                createResponseTopic(topic5.name, topic5.id, 9, 10, Errors.NONE)
+            ));
+
+        List<OffsetCommitResponseTopic> expectedTopics = new ArrayList<>();
+        expectedTopics.addAll(asList(
+            createResponseTopic(topicOne, topic1Id, partitionOne, partitionTwo, Errors.NONE),
+            createResponseTopic(topic6.name, topic6.id, 11, 12, Errors.NONE),
+            createResponseTopic(topicTwo, Uuid.ZERO_UUID, 3, 4, Errors.NONE),
+            createResponseTopic(topic3.name, topic3.id, 5, 6, Errors.NONE),
+            createResponseTopic(topic4.name, topic4.id, 7, 8, Errors.NONE),
+            createResponseTopic(topic5.name, topic5.id, 9, 10, Errors.NONE)
+        ));
+
+        OffsetCommitResponse response = builder.merge(coordinatorResults).build();
+        assertEquals(new OffsetCommitResponseData().setTopics(expectedTopics), response.data());
+    }
+
+    private static OffsetCommitResponseTopic createResponseTopic(
+        String topicName,
+        Uuid topicId,
+        int partition,
+        Errors error
+    ) {
+        return new OffsetCommitResponseTopic()
+            .setTopicId(topicId)
+            .setName(topicName)
+            .setPartitions(new ArrayList<>(asList(
+                new OffsetCommitResponsePartition()
+                    .setPartitionIndex(partition)
+                    .setErrorCode(error.code())
+            )));
+    }
+
+    private static OffsetCommitResponseTopic createResponseTopic(
+        String topicName,
+        Uuid topicId,
+        int firstPartition,
+        int secondPartition,
+        Errors error
+    ) {
+        OffsetCommitResponseTopic topic = createResponseTopic(topicName, topicId, firstPartition, error);
+        topic.partitions().add(new OffsetCommitResponsePartition()
+            .setPartitionIndex(secondPartition)
+            .setErrorCode(error.code()));
+
+        return topic;
     }
 
+    public static final class NameAndId {
```

Comment:
nit: TopicNameAndId?

Reply:
It is a bit weird to have this class defined here but I cannot think of a better place for now. Thoughts?